From 978aad85fdde665e009b84d4d54a05da8d183f8c Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 10:12:38 -0400
Subject: [fs] ext4: use truncate_pagecache() in collapse range

Message-id: <1414059176-6044-15-git-send-email-lczerner@redhat.com>
Patchwork-id: 98283
O-Subject: [RHEL 7.1 PATCH 14/32] ext4: use truncate_pagecache() in collapse range
Bugzilla: 1150171

BZ 1150171
https://bugzilla.redhat.com/show_bug.cgi?id=1150171

BREW 8149772
https://brewweb.devel.redhat.com/taskinfo?taskID=8149772

Upstream commit 694c793fc1ade0946149c5f8d43f71e0728c4e81

 We should be using truncate_pagecache() instead of
 truncate_pagecache_range() in the collapse range because we're
 truncating page cache from offset to the end of file.
 truncate_pagecache() also get rid of the private COWed pages from the
 range because we're going to shift the end of the file.

 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/extents.c b/fs/ext4/extents.c
index e512a17..ba25827 100644
--- a/fs/ext4/extents.c
+++ b/fs/ext4/extents.c
@@ -5432,7 +5432,7 @@ int ext4_collapse_range(struct inode *inode, loff_t offset, loff_t len)
   goto out_mutex;
  }
 
- truncate_pagecache_range(inode, offset, -1);
+ truncate_pagecache(inode, offset);
 
  /* Wait for existing dio to complete */
  ext4_inode_block_unlocked_dio(inode);
-- 
1.7.1